stl_to_obj/stl_to_tscn.py

```diff
@@ -0,0 +1,209 @@
+            self.texture_index[texture] = self.sub_resource_id
+            self.sub_resource_id += 1 
+
+    def add_obj_file(self, stl_file_extless, uv_map, mesh_compression, texture):
+        # reset bpy
+        bpy.ops.wm.read_factory_settings(use_empty=True)
+        # import .stl into bpy
+        bpy.ops.import_mesh.stl(filepath = f"{self.input_folder}/{stl_file_extless}.stl")
+        # Get all objects in selection
+        selection = bpy.context.selected_objects
+        # Deselect all objects
+        bpy.ops.object.select_all(action='DESELECT')
+
+        for obj in selection:
+            # Select each object
+            obj.select_set(True)
+            # Make it active
+            bpy.context.view_layer.objects.active = obj
+            # Toggle into Edit Mode
+            bpy.ops.object.mode_set(mode='EDIT')
+            # Select the geometry
+            bpy.ops.mesh.select_all(action='SELECT')
+
+            # Remove unnecessary faces 
+            if mesh_compression == "limited_dissolve": 
+                bpy.ops.mesh.dissolve_limited()
+            elif mesh_compression == "tris_to_quads":
+                bpy.ops.mesh.tris_convert_to_quads() 
+
+            # Call project operator to generate uv maps
+            try:
+                UV_MAPS[uv_map]()
+            except KeyError as e:
+                print(f"no uv map generated for {stl_file_extless} due to invalid uv_map")
+
+            # Toggle out of Edit Mode
+            bpy.ops.object.mode_set(mode='OBJECT')
+            # Deselect the object
+            obj.select_set(False)
+
+        # Export to obj
+        bpy.ops.wm.obj_export(filepath = f"{self.output_folder}/{stl_file_extless}.obj")
+        # remove the .mtl file
+        os.remove(f"{self.output_folder}/{stl_file_extless}.mtl")
+
+        # add to tscn file
+        self.nodes += f'[node name="{stl_file_extless}" type="MeshInstance" parent="."]\nmesh = ExtResource( {self.ext_resource_id} )' \
+                    + f'\nmaterial/0 = SubResource( {self.texture_index[texture]} )\n\n'
+        self.ext_resource += f'[ext_resource path="res://models/{self.output_folder}/{stl_file_extless}.obj" type="ArrayMesh" id={self.ext_resource_id}]\n'
+        self.ext_resource_id += 1
+
+    def add_collisions(self, stl_file_extless):
+        # generate fbx file from stl file to create collisions
+        fbx_file = f"{self.output_folder}/{stl_file_extless}.fbx"
+        scene = a3d.Scene.from_file(f"{self.input_folder}/{stl_file_extless}.stl")
+        scene.save(fbx_file)
+
+        # reset bpy
+        bpy.ops.wm.read_factory_settings(use_empty=True)
+        # import .fbx into bpy
+        bpy.ops.import_scene.fbx(filepath = fbx_file)
+
+        # rename all objects as -colonly
+        for obj_index in range(len(bpy.data.objects)):
+            # every re-name moves the object to the back of the list. 
+            bpy.data.objects[0].name = "Volume" + str(obj_index) + "-colonly"
```

Comment:
```suggestion
        for obj_index, _ in enumerate(bpy.data.objects):
            # every re-name moves the object to the back of the list. 
            bpy.data.objects[0].name = "Volume" + str(obj_index) + "-colonly"
```
More Pythonic to use `enumerate(...)` than `range(len(...))`. The underscore for the second item returned by enumerate (each object in `bpy.data.objects`) will treat that item as a throw-away since it's not needed.

You could alternatively use
```suggestion
        for obj_index, obj in enumerate(bpy.data.objects):
            # every re-name moves the object to the back of the list. 
            obj.name = "Volume" + str(obj_index) + "-colonly"
```
but that will preserve the order of `bpy.data.objects`. I'll take your word for it that each renaming moves the corresponding object to the back of the list rather than just rename the object in the first element of the list, while leaving the others unaffected.

Reply:
I applied that change in [this commit](https://github.com/cnerg/level-gen/pull/1/commits/8130e5b253c059ff030b6e482e59aec40f82f188) previously, but then @gonuke recommended the `range(len(bpy.data.objects))`.

Reply:
I liked the version with `range()` better than the version with `index, _ in enumerate()` because we don't use the `index` to access the entry in the list.  It seemed confusing to get an index and then reference the 0 element.

If the version with `index, obj in enumerate()` works - not sure because of the way we alter the list as part of this loop - then it's probably better.